From 939b1d6a20051f162aba48f60ddfe52b16fecee3 Mon Sep 17 00:00:00 2001
From: Noam.Lessner <nlessner@habana.ai>
Date: Fri, 05 Nov 2021 15:41:54 +0200
Subject: [PATCH] [SW-0] ppboot: iatu: update flag only if no skip

We need to update the boot status with iATU configured only
if we dont skip FW configuration.

Change-Id: Id29b05a0f8d80cc6b809933f35cd71f0707c7d8a
---

diff --git a/arch/gaudi2/arm/common.c b/arch/gaudi2/arm/common.c
index 6be8f53..f03b3ae 100644
--- a/arch/gaudi2/arm/common.c
+++ b/arch/gaudi2/arm/common.c
@@ -130,7 +130,9 @@
 
 void hl_plat_early_init(void)
 {
-	cpu_boot_dev_sts_set(CPU_BOOT_DEV_STS_FW_IATU_CONF_EN);
+	/* If FW skip disabled update boot status that iatu is configured */
+	if (!hl_skip_fw_cfg())
+		cpu_boot_dev_sts_set(CPU_BOOT_DEV_STS_FW_IATU_CONF_EN);
 
 	gaudi2_axi_drain_ena();
 
